sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -379,6 +381,33 @@ class Analyzer(catalog: Catalog,
     }
   }
 
+  object ResolveSortAlias extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case s @ Sort(ordering, global, a @ Aggregate(grouping, aggs, child))
+          if !s.resolved && a.resolved =>
+        val missingInAggs = new ArrayBuffer[NamedExpression]()
+        val aliasedOrdering = ordering.zipWithIndex.map{
+          case (o, i) => {
+            o.child match {
+              case aggOrSub @ (_: AggregateExpression | _: Substring) => {
+                val aliasName = aggOrSub.nodeName + i
+                missingInAggs += Alias(aggOrSub, aliasName)()
+                SortOrder(UnresolvedAttribute(aliasName), o.direction)
```

Comment:
Its not safe to create `UnresolvedAttributes` in query plans.  What if the user happens to have a name that conflicts?  Instead you should call `.toAttribute` on the `Alias` you are creating.  This will always ensure that you have a globally unique identifier that cannot have conflicts.